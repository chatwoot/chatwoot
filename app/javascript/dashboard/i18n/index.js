import ar from './locale/ar';
import ca from './locale/ca';
import cs from './locale/cs';
import da from './locale/da';
import de from './locale/de';
import el from './locale/el';
import en from './locale/en';
import es from './locale/es';
import fa from './locale/fa';
import fi from './locale/fi';
import fr from './locale/fr';
import he from './locale/he';
import hi from './locale/hi';
import hu from './locale/hu';
import id from './locale/id';
import it from './locale/it';
import ja from './locale/ja';
import ko from './locale/ko';
import lv from './locale/lv';
import ml from './locale/ml';
import nl from './locale/nl';
import no from './locale/no';
import pl from './locale/pl';
import pt from './locale/pt';
import pt_BR from './locale/pt_BR';
import ro from './locale/ro';
import ru from './locale/ru';
import sk from './locale/sk';
import sv from './locale/sv';
import ta from './locale/ta';
import th from './locale/th';
import tr from './locale/tr';
import uk from './locale/uk';
import vi from './locale/vi';
import zh_CN from './locale/zh_CN';
import zh_TW from './locale/zh_TW';
import is from './locale/is';

export default {
  ar,
  ca,
  cs,
  da,
  de,
  el,
  en,
  es,
  fa,
  fi,
  fr,
  he,
  hi,
  hu,
  id,
  it,
  ja,
  ko,
  ml,
  lv,
  nl,
  no,
  pl,
  pt_BR,
  pt,
  ro,
  ru,
  sk,
  sv,
  ta,
  th,
  tr,
  uk,
  vi,
  zh_CN,
  zh_TW,
  is,
};
