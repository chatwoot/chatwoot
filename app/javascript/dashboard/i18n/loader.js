export async function setLocale(i18n, locale) {
  const { default: messages } = await import(`./locale/${locale}/index.js`);

  i18n.setLocaleMessage(locale, messages);
  i18n.locale.value = locale;
}
