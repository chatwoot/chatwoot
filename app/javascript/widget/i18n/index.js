import ar from './locale/ar.json';
import ca from './locale/ca.json';
import cs from './locale/cs.json';
import da from './locale/da.json';
import de from './locale/de.json';
import el from './locale/el.json';
import en from './locale/en.json';
import es from './locale/es.json';
import fa from './locale/fa.json';
import fi from './locale/fi.json';
import fr from './locale/fr.json';
import he from './locale/he.json';
import hi from './locale/hi.json';
import hu from './locale/hu.json';
import id from './locale/id.json';
import is from './locale/is.json';
import it from './locale/it.json';
import ja from './locale/ja.json';
import ko from './locale/ko.json';
import lv from './locale/lv.json';
import ml from './locale/ml.json';
import nl from './locale/nl.json';
import no from './locale/no.json';
import pl from './locale/pl.json';
import pt from './locale/pt.json';
import pt_BR from './locale/pt_BR.json';
import ro from './locale/ro.json';
import ru from './locale/ru.json';
import sk from './locale/sk.json';
import sv from './locale/sv.json';
import ta from './locale/ta.json';
import th from './locale/th.json';
import tr from './locale/tr.json';
import uk from './locale/uk.json';
import vi from './locale/vi.json';
import zh_CN from './locale/zh_CN.json';
import zh_TW from './locale/zh_TW.json';

export default {
  ar,
  ca,
  cs,
  da,
  de,
  el,
  en,
  es,
  fa,
  fi,
  fr,
  he,
  hi,
  hu,
  id,
  is,
  it,
  ja,
  ko,
  lv,
  ml,
  nl,
  no,
  pl,
  pt_BR,
  pt,
  ro,
  ru,
  sk,
  sv,
  ta,
  th,
  tr,
  uk,
  vi,
  zh_CN,
  zh_TW,
};
