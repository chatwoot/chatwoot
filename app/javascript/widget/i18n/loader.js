export async function setLocale(i18n, locale) {
  const { default: messages } = await import(`./locale/${locale}.json`);

  i18n.setLocaleMessage(locale, messages);
  i18n.locale.value = locale;
}
