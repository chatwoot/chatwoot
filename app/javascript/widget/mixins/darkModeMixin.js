import { mapGetters } from 'vuex';

export default {
  computed: {
    ...mapGetters({ darkMode: 'appConfig/darkMode' }),
  },
  methods: {
    $dm(light, dark) {
      if (this.darkMode === 'light') {
        return light;
      }
      if (this.darkMode === 'dark') {
        return dark;
      }
      return light + ' ' + dark;
    },
  },
};
