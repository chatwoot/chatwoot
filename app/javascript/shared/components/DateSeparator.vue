<template>
  <div
    class="date--separator"
    :class="$dm('text-slate-700', 'dark:text-slate-200')"
  >
    {{ formattedDate }}
  </div>
</template>

<script>
import { formatDate } from 'shared/helpers/DateHelper';
import darkModeMixin from 'widget/mixins/darkModeMixin.js';

export default {
  mixins: [darkModeMixin],
  props: {
    date: {
      type: String,
      required: true,
    },
  },
  computed: {
    formattedDate() {
      return formatDate({
        date: this.date,
        todayText: this.$t('TODAY'),
        yesterdayText: this.$t('YESTERDAY'),
      });
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~widget/assets/scss/variables';

.date--separator {
  font-size: $font-size-default;
  height: 50px;
  line-height: 50px;
  position: relative;
  text-align: center;
  width: 100%;
}

.date--separator::before,
.date--separator::after {
  background-color: $color-border;
  content: '';
  height: 1px;
  position: absolute;
  top: 24px;
  width: calc((100% - 120px) / 2);
}

.date--separator::before {
  left: 0;
}

.date--separator::after {
  right: 0;
}
</style>
