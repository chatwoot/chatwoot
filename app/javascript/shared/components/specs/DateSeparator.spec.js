import DateSeparator from '../DateSeparator';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import VueI18n from 'vue-i18n';
import darkModeMixin from 'widget/mixins/darkModeMixin.js';
const localVue = createLocalVue();
import i18n from 'dashboard/i18n';
localVue.use(Vuex);
localVue.use(VueI18n);

const i18nConfig = new VueI18n({
  locale: 'en',
  messages: i18n,
});

describe('dateSeparator', () => {
  let store = null;
  let actions = null;
  let modules = null;
  let dateSeparator = null;

  beforeEach(() => {
    actions = {};

    modules = {
      auth: {
        getters: {
          'appConfig/darkMode': () => 'light',
        },
      },
    };
    store = new Vuex.Store({
      actions,
      modules,
    });

    dateSeparator = shallowMount(DateSeparator, {
      store,
      localVue,
      propsData: {
        date: 'Nov 18, 2019',
      },
      i18n: i18nConfig,
      mixins: [darkModeMixin],
    });
  });

  it('date separator snapshot', () => {
    expect(dateSeparator.vm).toBeTruthy();
    expect(dateSeparator.element).toMatchSnapshot();
  });
});
