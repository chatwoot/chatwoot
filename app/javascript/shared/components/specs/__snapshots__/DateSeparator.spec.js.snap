// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateSeparator date separator snapshot 1`] = `
<div
  class="date--separator text-slate-700"
>
  
  Nov 18, 2019

</div>
`;
